### ISSUE TITLE
Vim9: incorrect documentation regarding prefix of script-local variables in :def function in legacy script

### ISSUE THREAD

**Issue by lacygoill** [vim/vim#6773]
**Describe the bug**

The help after `:h :enddef` says that the `s:` prefix is necessary to access a script-local variable from a `:def` function in a legacy script:

https://github.com/vim/vim/blob/1b04ce2d400fda97410a961288c496bd8f445a9c/runtime/doc/vim9.txt#L540-L542

That doesn't seem to be the case.

**To Reproduce**

Run this shell command:

    vim -Nu NONE -S <(cat <<'EOF'
        let s:var = 123
        def Func()
            echo var
        enddef
        call Func()
    EOF
    )

No error is raised.  `123` is correctly echo'ed.

**Expected behavior**

If the documentation is correct, then an error should be raised.

**Environment**

 - Vim version: 8.2 Included patches: 1-1505
 - OS: Ubuntu 16.04.7 LTS
 - Terminal: XTerm(358)

**Additional context**

IMO, the current behavior is good; as a user, I would expect that what matters is the immediate outer context.  Here, the immediate outer context is a `:def` function where the Vim9 syntax can be used; and when the Vim9 syntax can be used, `s:` can be omitted.

I would rather keep the current behavior and update the documentation, but maybe that's not possible; I don't know the implementation details.

**Comment by lacygoill** [comment]
If it is not disallowed to define a script-local variable in a `:def` function in a legacy script, then I think the documentation should be updated with a patch similar to this:
```diff
diff --git a/runtime/doc/vim9.txt b/runtime/doc/vim9.txt
index fa7febc4c..aebd74b24 100644
--- a/runtime/doc/vim9.txt
+++ b/runtime/doc/vim9.txt
@@ -539,7 +539,8 @@ If the script the function is defined in is Vim9 script, then script-local
 variables can be accessed without the "s:" prefix.  They must be defined
 before the function is compiled.  If the script the function is defined in is
 legacy script, then script-local variables must be accessed with the "s:"
-prefix.
+prefix, if they were defined in a `:def` function; otherwise, if they were
+defined at the script level, "s:" can be omitted.
 
 						*:defc* *:defcompile*
 :defc[ompile]		Compile functions defined in the current script that
```
Although, this makes the paragraph longer, and it gets harder to build an intuition around it.  IMO, this suggests that it should be disallowed.

**Comment by lacygoill** [comment]
Fixed by [8.2.1510](https://github.com/vim/vim/commit/5390099a9733f7952a612670693dd4ebf9e0e178).
